Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp

```diff
@@ -1571,6 +1583,15 @@ bool GraphicsContextCG::canUseShadowBlur() const
     return (renderingMode() == RenderingMode::Unaccelerated) && hasBlurredShadow() && !m_state.shadowsIgnoreTransforms();
 }
 
+bool GraphicsContextCG::commitDraws()
+{
+    if (m_hasDrawnSinceLastCommit) {
+        m_hasDrawnSinceLastCommit = false;
```

Comment:
It seems slightly sad that a caller could still have access to the `CGContextRef` obtained from `platformContext` when this is called, and then add new draw calls later that wouldn't be tracked.

It would be nice if the public API returned some sort of 'borrowed' object, that we could assert has been returned/destroyed before this was called.

I guess that still doesn't help too much, since there are still other ways for external users to have the `CGContextRef` (like from the IOSurface), and thus we have to still manually determine external reasons to flush (or that flushing has been done for us).

Without large changes to make this a much more closed abstraction around the `CGContext`, then this seems like the best way can do as a 'have there been any draw calls visible to this object, since the last time this was called'.

Not sure what the resolution is for the naming discussion, but the approach seems reasonable to me.

Reply:
>It would be nice if the public API returned some sort of 'borrowed' object, that we could assert has been returned/destroyed before this was called.

This is explored in #11354 [11354](https://github.com/WebKit/WebKit/pull/11354)

> I guess that still doesn't help too much, since there are still other ways for external users to have the CGContextRef (like from the IOSurface), and thus we have to still manually determine external reasons to flush (or that flushing has been done for us).

I think this is a bug in IOSurface -- there's no use-case for such a construct as to store the stale or random CGContext from the IOSurface. IOSurface::ensurePlatformContext() should be renamed as IOSurface::createPlatformContext().